### PR TITLE
tests: Adds tests for properties "leak"

### PR DESCRIPTION
**Pull request by breakreturn**
Test pinpoints the behavior where the proxy can't
disallow calls to e.g. GetAll based on the interface
argument, and thus possibly "leak" properties from
an otherwise disallowed interface.

Signed-off-by: Joakim Gross <joakim.gross@pelagicore.com>

### CODE REVIEW

File: component-test/test_dbus_proxy.py [hunk 1]
```diff
@@ -816,6 +816,102 @@ class TestProxyBehaviorForOrgFreedesktopPropertiesIface(object):
...
+        """.format(**{
+            "props_iface": dbus.PROPERTIES_IFACE,
+            "opath_1": stubs.OPATH_1,
+            "iface1_1": stubs.TestInterface1_1
+        })
+
+        dbus_proxy.set_config(config)
+
+        # Assert the interface that we later get properties from is
+        # otherwise not allowed by be reached by the proxy.
```
Comment (thenor) [inline component-test/test_dbus_proxy.py]: I think there is a typo in this text. "... allowed **by** be reached ..." ?

File: component-test/service_stubs.py [hunk 2]
```diff
@@ -44,21 +44,33 @@
...
-    Object1 also implements GetAll and Get on the "org.freedesktop.DBus.Properties"
-    interface. Object2 implements GetAll on the same interface.
+        org.freedesktop.DBus.Properties
+            GetAll
+
+    Object1 will return properties based on the interface argument passed
+    to GetAll. This is to mimic a typical service that respects the argument
+    and returns properties based on it. This behavior is used by
+    component tests to assert the behavior of dbus-proxy when used in this
+    context. E.g, dbus-proxy will not disallow properties on interface X
```
Comment (thenor) [inline component-test/service_stubs.py]: I think there is a typo here: "E.g**,** ..."

**Review by thenor — CHANGES_REQUESTED** [review]
Just found two small typos. Otherwise a-OK.

**Review by frznlogic — APPROVED** [review]

File: component-test/service_stubs.py [hunk 3]
```diff
@@ -103,19 +127,23 @@ def __init__(self, bus):
 
         # Some nonsensical properties so there is something to return
         # from GetAll
-        self.__properties = {"MyKey1": "my_value_1", "MyKey2": "my_value_2"}
+        self.__properties = dict()
+        ifaces = [TestInterface1_1, TestInterface1_1_2]
+        for iface in ifaces:
+            self.__properties[iface] = {PROP_KEY_1 + iface: PROP_VALUE_1 + iface,
+                                        PROP_KEY_2 + iface: PROP_VALUE_2 + iface}
```
Comment (frznlogic) [inline component-test/service_stubs.py:134]: I understand the code, but IMHO, the old code is actually easier to read/understand for me ;).

Comment (breakreturn) [reply component-test/service_stubs.py:134]: Yes it was, but it also did something else. Would you propose to have the new functionality hard-coded?

**Review by thenor — APPROVED** [review]

File: component-test/service_stubs.py (hunk 3 above)
Comment (frznlogic) [reply component-test/service_stubs.py:134]: Definitely not, it was mostly a comment on the discussion we had over the last few weeks on the topic of hardcoded variables etc ;). I digress.

**Pull request merged**